packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -0,0 +1,245 @@
+import type { ReactElement } from 'react';
+import type { HeadingProps, TextProps, TitleProps } from '../Typography';
+import {
+  amountFontSizes,
+  currencyAbbreviationsMapping,
+  currencyLocaleMapping,
+  currencyPrefixMapping,
+  affixFontSizes,
+} from './amountTokens';
+import { BaseText } from '~components/Typography/BaseText';
+import type { Feedback } from '~tokens/theme/theme';
+import type { BaseTextProps } from '~components/Typography/BaseText/types';
+import BaseBox from '~components/Box/BaseBox';
+import type { TestID } from '~src/_helpers/types';
+import { metaAttribute, MetaConstants } from '~utils';
+
+type Currency = 'INR' | 'MYR';
+
+type AmountProps = {
+  /**
+   * The value to be rendered within the component.
+   *
+   */
+  value: number;
+  /**
+   * Sets the intent of the amount.
+   *
+   * @default undefined
+   */
+  intent?: Exclude<Feedback, 'neutral'>;
+  /**
+   * Sets the size of the amount.
+   *
+   * @default 'body-medium'
+   */
+  size?:
+    | `title-${NonNullable<Exclude<TitleProps['size'], 'large'>>}`
+    | `heading-${NonNullable<Exclude<HeadingProps<{ variant: 'regular' }>['size'], 'medium'>>}`
+    | `heading-${NonNullable<
+        Exclude<HeadingProps<{ variant: 'regular' }>['size'], 'medium'>
+      >}-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['weight'], 'regular'>>}`
+    | `body-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['size'], 'xsmall'>>}`
+    | `body-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['size'], 'xsmall'>>}-${NonNullable<
+        Exclude<TextProps<{ variant: 'body' }>['weight'], 'regular'>
+      >}`;
+  /**
+   * Indicates what the suffix of amount should be
+   *
+   * @default 'decimals'
+   */
+  suffix?: 'decimals' | 'none' | 'humanize';
+  /**
+   * in amount by making the prefix symbol and decimal digits s
+   *
+   * @default true
+   */
+  isAffixSubtle?: true | false;
+  /**
+   * Prefix to be shown before the amount value. The prefix can be either a currency symbol or a currency code.
+   *
+   * @default 'currency-symbol'
+   */
+  prefix?: 'currency-symbol' | 'currency-code';
+} & TestID;
+
+type ColorProps = {
+  amountValueColor: BaseTextProps['color'];
+  affixColor: BaseTextProps['color'];
+};
+
+const getTextColorProps = ({ intent }: { intent: AmountProps['intent'] }): ColorProps => {
+  const props: ColorProps = {
+    amountValueColor: 'surface.text.normal.lowContrast',
+    affixColor: 'surface.text.muted.lowContrast',
+  };
+  if (!intent && intent !== 'neutral') return props;
```

Comment:
Can we also throw an error when someone passes `neutral` as an intent? Otherwise people might pass neutral in JS projects and raise a bug that it doesn't work.
```tsx
if(intent==='neutral')
{
  throw new Error("[Blade Amount]: `neutral` intent is not support.")
}
```